src/coprocessor/dag/rpn_expr/impl_time.rs

```diff
@@ -97,28 +101,32 @@ mod tests {
             assert_eq!(output, expect, "{:?} {:?}", date, format);
         }
 
-        //        // TODO: pass this test after refactoring the issue #3953 is fixed.
-        //        {
-        //            // test for warning context
-        //            use std::sync::Arc;
-        //            use crate::coprocessor::dag::expr::{EvalConfig, EvalContext, Flag, SqlMode};
-        //
-        //            let format = Some("abc%b %M %m %c %D %d %e %j").map(|s| s.as_bytes().to_vec());
-        //            let time = Some(DateTime::parse_utc_datetime("0000-00-00 00:00:00", 6).unwrap());
-        //
-        //            let mut cfg = EvalConfig::new();
-        //            cfg.set_flag(Flag::IN_UPDATE_OR_DELETE_STMT)
-        //                .set_sql_mode(m::ERROR_FOR_DIVISION_BY_ZERO | SqlMode::STRICT_ALL_TABLES);
-        //            let ctx = EvalContext::new(Arc::new(cfg));
-        //
-        //            let output = RpnFnScalarEvaluator::new()
-        //                .context(ctx)
-        //                .push_param(time.clone())
-        //                .push_param(format)
-        //                .evaluate::<Bytes>(ScalarFuncSig::DateFormatSig)
-        //                .unwrap();
-        //            assert_eq!(output, None, "{:?} {:?}", time, "");
-        //        }
+        let cases: Vec<(Option<&str>, Option<&str>)> = vec![
+            //            // TODO: pass this test after refactoring the issue #3953 is fixed.
+            //            (
+            //                Some("abc%b %M %m %c %D %d %e %j"),
+            //                Some("0000-00-00 00:00:00"),
+            //            ),
+            (None, None),
+        ];
+
+        for (format, time) in cases {
```

Comment:
No need to be a loop for the None -> None case. Other test cases are not suitable for these flags.

Reply:
> No need to be a loop for the None -> None case. Other test cases are not suitable for these flags.

What about test for `"0000-00-00 00:00:00"`?

Reply:
> No need to be a loop for the None -> None case. Other test cases are not suitable for these flags.

Oh, I just made a mistake. `t.invalid_zero` should return error in strict mode.
The tidb code is here: https://github.com/pingcap/tidb/blob/a737d26e4f126cbd6cff6872e10003c66280d423/expression/errors.go